llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -362,6 +364,34 @@ static bool expandClampIntrinsic(CallInst *Orig, Intrinsic::ID ClampIntrinsic) {
   return true;
 }
 
+static bool expandSaturateIntrinsic(CallInst *SaturateCall) {
+  FunctionType *FT = SaturateCall->getFunctionType();
+  Type *FTRetTy = FT->getReturnType();
+  assert(FTRetTy == FT->getParamType(0) &&
```

Comment:
It looks like you are trying to do scalarization via this pass, but we haven't been doing that thus far. I wanted to take care of scalarization in a different pass because doing it here means we have to do it for each intrinsic which I want to avoid. My feeling is scalarization should be uniform across all intrinsics unless there are special cases like `dot2`, `dot3`, `dot4`.

Reply:
> It looks like you are trying to do scalarization via this pass, but we haven't been doing that thus far. I wanted to take care of scalarization in a different pass because doing it here means we have to do it for each intrinsic which I want to avoid. My feeling is scalarization should be uniform across all intrinsics unless there are special cases like `dot2`, `dot3`, `dot4`.

The functionality can be moved out of here once such a scalarization pass is implemented.

Reply:
I think we should drop this for now. We likely won't be preserving any intrinsic specific behaviors.

Reply:
> I think we should drop this for now. We likely won't be preserving any intrinsic specific behaviors.

Pushed a change dropping scalarization functionality and associated tests.